### ISSUE TITLE
Option to not automatically compile in source file associated with test file?

### ISSUE THREAD

**Issue by laurensmiers** [ThrowTheSwitch/Ceedling#242]
Hey all,

Title may be a bit confusing but bear with me.
At my project, we are including the c-file-under-test in our test-file.
Disscussion is possible on if including the c-file is the best way, but sometimes there are static functions where it doesn't make sense to split them of in a separate file/module/.... or we have cases where we want to trigger a corner case on a static function.
And we would like to avoid using a define like STATIC or something to expose the functions/data structures in our tests.

The issue is that ceedling automatically compiles in the c-file-under-test when compiling the tests and this results in linker errors with multiple definitions, etc.
We worked around it by not including the c-files-under-test in the files section:
```

:files:
  :source:
    - -: ./services/src/*.c
```

But this means that sometimes when using "TEST_FILE("mymodule.c")" in a test, it can't find the source file because it is removed with the snippet above.

I was just wondering if there is an option (in project settings or whatever), to not automatically compile the c-file-under-test in (since it is already included in the test-file).

KR,

Laurens

**Comment by mmmack678** [comment]
Hi everyone,

I have the same problem. Would it be a possibility to add a macro similar to TEST_FILE, that instead of specifying additional compilation units, tells Ceedling to omit this when linking?

E.g NO_LINK("mymodule.c") would be like #include "mymodule.c" but prevent Ceedling from using mymodule.o when linking.

Best regards,
Michael

**Comment by mvandervoord** [comment]
An excellent suggestions, @mmmack678

**Comment by mnkp** [comment]
Hi all,

I also would welcome an ability to include c-file-under-test in the test-file, however I'm not so sure if a NO_LINK macro should be the only way to achieve it. A macro like that may be useful for its own merits, but that doesn't change the fact that the current behaviour feels wrong. If user includes a .c file in their test-file there is no reason to compile and link this file separately. I reckon most of us expected that including a .c file should simply work and were surprised to discover that it didn't. 

To sum up, the current behaviour feels like a bug not a feature. Only .c files which corresponding .h file was included should be automatically compiled and linked.

In any case, thanks @mvandervoord for a great unit test framework.

Best regards,
Piotr
